proxy/src/telemetry/sensor/http.rs

```diff
@@ -387,36 +409,154 @@ where
             Ok(Async::NotReady) => Ok(Async::NotReady),
             Ok(Async::Ready(trls)) => {
                 if let Some(i) = self.inner.take() {
-                    let ResponseBodyInner {
-                        ctx,
-                        mut handle,
-                        request_open,
-                        response_open,
-                        bytes_sent,
-                        frames_sent,
-                    } = i;
-
-                    handle.send(|| {
-                        let grpc_status = trls.as_ref()
-                            .and_then(|t| t.get(GRPC_STATUS))
-                            .and_then(|v| v.to_str().ok())
-                            .and_then(|s| s.parse::<u32>().ok());
-
-                        event::Event::StreamResponseEnd(
-                            Arc::clone(&ctx),
-                            event::StreamResponseEnd {
-                                grpc_status,
-                                since_request_open: request_open.elapsed(),
-                                since_response_open: response_open.elapsed(),
-                                bytes_sent,
-                                frames_sent,
-                            },
-                        )
-                    })
+                    let grpc_status = trls.as_ref()
+                        .and_then(|t| t.get(GRPC_STATUS))
+                        .and_then(|v| v.to_str().ok())
+                        .and_then(|s| s.parse::<u32>().ok());
+                    i.end(grpc_status);
```

Comment:
It also looks like requests don't send trailers at all in gRPC, so the request duration would actually be potentially earlier, when the data stream ends. I don't know if that distinction in durations is important or not.

Reply:
@adleong, do you think the difference in durations here matters?

Reply:
@seanmonstar, @adleong says that if the difference in duration is on the order of nanoseconds here, we probably don't care from a metrics perspective. Do you think that it's likely for the difference in durations to be significant?

Reply:
Then perhaps not. I do think that recording a stream has ended in `poll_trailers` is a little brittle, since polling trailers makes no sense in HTTP/1, but I can tackle that some day in the future.

Reply:
Hmm. I'm happy to continue making changes to this code as well, if you have any suggestions.